packages/static-website/src/origin-access-control.ts

```diff
@@ -0,0 +1,125 @@
+/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0 */
+import * as path from "path";
+import { ArnFormat, CustomResource, Stack } from "aws-cdk-lib";
+import {
+  CfnDistribution,
+  CfnOriginAccessControl,
+  Distribution,
+} from "aws-cdk-lib/aws-cloudfront";
+import { Effect, PolicyStatement } from "aws-cdk-lib/aws-iam";
+import { NodejsFunction } from "aws-cdk-lib/aws-lambda-nodejs";
+import { CfnBucketPolicy, IBucket } from "aws-cdk-lib/aws-s3";
+import { Provider } from "aws-cdk-lib/custom-resources";
+import { Construct } from "constructs";
+
+/**
+ * Props for the Origin Access Control.
+ */
+export interface OriginAccessControlProps {
+  /**
+   * The Distribution to apply OAC to.
+   */
+  readonly distribution: Distribution;
+
+  /**
+   * The bucket to apply OAC to.
+   */
+  readonly bucket: IBucket;
+}
+
+/**
+ * Creates and associates a OriginAccessControl to a distribution and bucket.
+ *
+ * @stability experimental
+ */
+export class OriginAccessControl extends Construct {
+  public readonly oac;
+  constructor(scope: Construct, id: string, props: OriginAccessControlProps) {
+    super(scope, id);
+
+    const { distribution, bucket } = props;
+    const stack = Stack.of(this);
+    this.oac = new CfnOriginAccessControl(this, `${id}-OAC`, {
+      originAccessControlConfig: {
+        name: `${stack.stackName}_OAC`,
```

Comment:
If we need to hardcode a name it's best to include something like `this.node.addr.slice(-8)` for some additional uniqueness, eg if we need to add 2 websites in the same stack :)